src/abstractdatatable/reshape.jl

```diff
@@ -193,9 +193,23 @@ function unstack(dt::AbstractDataTable, rowkey::Int, colkey::Int, value::Int)
     # `rowkey` integer indicating which column to place along rows
     # `colkey` integer indicating which column to place along column headers
     # `value` integer indicating which column has values
-    refkeycol = NullableCategoricalArray(dt[rowkey])
+    if typeof(dt[rowkey]) <: CategoricalVector
```

Comment:
I've just pushed a commit to the nl/null CategoricalArrays branch which should hopefully fix this (still need to add some tests though).